### PR TITLE
Prevent from unused variable warning when traces are disabled

### PR DESCRIPTION
**Pull request by ktrzcinx**
Moreover, improve solution from https://github.com/thesofproject/sof/pull/2022 where thread started in https://github.com/thesofproject/sof/issues/1999.

### CODE REVIEW

File: src/include/sof/trace/preproc.h [hunk 1]
```diff
@@ -157,4 +157,9 @@
 #define META_FUNC_WITH_VARARGS(prefix, postfix, return_t, args)\
 		return_t META_CONCAT(prefix, postfix) (args)
 
+/* counteract compilator warning about unused variables */
```
Comment (paulstelian97) [inline src/include/sof/trace/preproc.h]: s/compilator/compiler

File: src/lib/wait.c [hunk 2]
```diff
@@ -17,6 +17,7 @@
 #include <user/trace.h>
 #include <errno.h>
 #include <stdint.h>
+#include <inttypes.h>
```
Comment (jajanusz) [inline src/lib/wait.c]: why do you include `inttypes`?

Comment (paulstelian97) [reply src/lib/wait.c]: PRIu64 below.

File: src/lib/wait.c [hunk 3]
```diff
@@ -41,8 +42,8 @@ int wait_for_completion_timeout(completion_t *comp)
 	}
 
 	/* timeout */
-	trace_error_value(c->timeout);
-	trace_error_value(c->complete);
+	trace_error(TRACE_CLASS_WAIT, "wait_for_completion_timeout, timeout: " /
```
Comment (paulstelian97) [inline src/lib/wait.c]: No need for "/" at the end of the line (also compiler error :) )

Comment (paulstelian97) [reply src/lib/wait.c]: PRIu64 is something like "%llu" AKA a string.

File: src/include/sof/trace/trace.h [hunk 4]
```diff
@@ -198,14 +198,6 @@ void trace_init(struct sof *sof);
 	_trace_event_atomic_with_ids(class, id_0, id_1, 1, format,	\
 				     ##__VA_ARGS__)
 
-/* component tracing builder macro */
```
Comment (lgirdwood) [inline src/include/sof/trace/trace.h]: The commit message explains why we make the change, but it does not explain what the change is i.e. are we moving trace defs around or are we modifiying the definitions ?

File: src/audio/detect_test.c [hunk 5]
```diff
@@ -544,7 +544,7 @@ static int test_keyword_get_config(struct comp_dev *dev,
 
 	/* Copy back to user space */
 	bs = cd->config.size;
-	trace_value(bs);
+	trace_keyword("value of bs: %u", bs);
```
Comment (lgirdwood) [inline src/audio/detect_test.c]: What is "bs" ? Best to use "block size " if bs represent block size.

File: src/include/sof/trace/trace.h [hunk 6]
```diff
@@ -166,6 +166,9 @@ void trace_on(void);
 void trace_off(void);
 void trace_init(struct sof *sof);
 
+#define trace_unused(class, id_0, id_1, format, ...) \
```
Comment (lgirdwood) [inline src/include/sof/trace/trace.h]: Commit message spelling.

File: src/audio/detect_test.c (hunk 5 above)
Comment (ktrzcinx) [reply src/audio/detect_test.c]: changed

File: src/include/sof/trace/trace.h (hunk 4 above)
Comment (ktrzcinx) [reply src/include/sof/trace/trace.h]: added

File: src/include/sof/trace/trace.h (hunk 6 above)
Comment (ktrzcinx) [reply src/include/sof/trace/trace.h]: should be better

File: src/lib/wait.c (hunk 3 above)
Comment (ktrzcinx) [reply src/lib/wait.c]: ok

File: src/lib/wait.c (hunk 2 above)
Comment (ktrzcinx) [reply src/lib/wait.c]: right

File: src/include/sof/trace/preproc.h (hunk 1 above)
Comment (ktrzcinx) [reply src/include/sof/trace/preproc.h]: changed

**Comment by ktrzcinx** [comment]
SOFCI TEST

**Pull request merged**